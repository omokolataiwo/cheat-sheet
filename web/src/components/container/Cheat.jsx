import React from 'react';
import propTypes from 'prop-types';
import { CopyToClipboard } from 'react-copy-to-clipboard';
import Tooltip from 'rc-tooltip';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Cheat = ({
  showActionBar,
  cheat: {
    description, line, _id: id, favorite
  }, onAddToFavorite
}) => (
  <div className="cheat">
    <p>{description}</p>
    <CopyToClipboard text={line}>
      <Tooltip placement="left" trigger={['click']} overlay={<span>Copied!</span>}>
        <p className="line">
          {`$ ${line}`}
        </p>
      </Tooltip>
    </CopyToClipboard>
    {showActionBar && localStorage.getItem('token') && <FontAwesomeIcon className={favorite ? 'favorite' : ''} icon="heart" onClick={() => onAddToFavorite(id)} />}
  </div>
);

Cheat.propTypes = {
  cheat: propTypes.shape({
    description: propTypes.string.isRequired,
    line: propTypes.string.isRequired
  }).isRequired,
  onAddToFavorite: propTypes.func.isRequired
};

export default Cheat;
