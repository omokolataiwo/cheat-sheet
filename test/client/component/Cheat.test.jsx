import React from 'react';
import { shallow } from 'enzyme';
import Cheat from '../../../web/src/components/container/Cheat';

const props = {
  cheat: { description: 'init repository', line: 'git init' },
  onAddToFavorite: jest.fn(() => {})
};
const wrapper = shallow(<Cheat {...props} />);

describe('Cheat Component', () => {
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
