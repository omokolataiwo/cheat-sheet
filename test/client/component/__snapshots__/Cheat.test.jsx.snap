// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cheat Component should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cheat
    cheat={
      Object {
        "description": "init repository",
        "line": "git init",
      }
    }
    onAddToFavorite={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          init repository
        </p>,
        <CopyToClipboard
          text="git init"
        >
          <Tooltip
            align={Object {}}
            arrowContent={null}
            destroyTooltipOnHide={false}
            mouseEnterDelay={0}
            mouseLeaveDelay={0.1}
            overlay={
              <span>
                Copied!
              </span>
            }
            placement="left"
            prefixCls="rc-tooltip"
            trigger={
              Array [
                "click",
              ]
            }
          >
            <p
              className="line"
            >
              $ git init
            </p>
          </Tooltip>
        </CopyToClipboard>,
        undefined,
      ],
      "className": "cheat",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "init repository",
        },
        "ref": null,
        "rendered": "init repository",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Tooltip
            align={Object {}}
            arrowContent={null}
            destroyTooltipOnHide={false}
            mouseEnterDelay={0}
            mouseLeaveDelay={0.1}
            overlay={
              <span>
                Copied!
              </span>
            }
            placement="left"
            prefixCls="rc-tooltip"
            trigger={
              Array [
                "click",
              ]
            }
          >
            <p
              className="line"
            >
              $ git init
            </p>
          </Tooltip>,
          "onCopy": undefined,
          "options": undefined,
          "text": "git init",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": Object {},
            "arrowContent": null,
            "children": <p
              className="line"
            >
              $ git init
            </p>,
            "destroyTooltipOnHide": false,
            "mouseEnterDelay": 0,
            "mouseLeaveDelay": 0.1,
            "overlay": <span>
              Copied!
            </span>,
            "placement": "left",
            "prefixCls": "rc-tooltip",
            "trigger": Array [
              "click",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "$ git init",
              "className": "line",
            },
            "ref": null,
            "rendered": "$ git init",
            "type": "p",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            init repository
          </p>,
          <CopyToClipboard
            text="git init"
          >
            <Tooltip
              align={Object {}}
              arrowContent={null}
              destroyTooltipOnHide={false}
              mouseEnterDelay={0}
              mouseLeaveDelay={0.1}
              overlay={
                <span>
                  Copied!
                </span>
              }
              placement="left"
              prefixCls="rc-tooltip"
              trigger={
                Array [
                  "click",
                ]
              }
            >
              <p
                className="line"
              >
                $ git init
              </p>
            </Tooltip>
          </CopyToClipboard>,
          undefined,
        ],
        "className": "cheat",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "init repository",
          },
          "ref": null,
          "rendered": "init repository",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Tooltip
              align={Object {}}
              arrowContent={null}
              destroyTooltipOnHide={false}
              mouseEnterDelay={0}
              mouseLeaveDelay={0.1}
              overlay={
                <span>
                  Copied!
                </span>
              }
              placement="left"
              prefixCls="rc-tooltip"
              trigger={
                Array [
                  "click",
                ]
              }
            >
              <p
                className="line"
              >
                $ git init
              </p>
            </Tooltip>,
            "onCopy": undefined,
            "options": undefined,
            "text": "git init",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "align": Object {},
              "arrowContent": null,
              "children": <p
                className="line"
              >
                $ git init
              </p>,
              "destroyTooltipOnHide": false,
              "mouseEnterDelay": 0,
              "mouseLeaveDelay": 0.1,
              "overlay": <span>
                Copied!
              </span>,
              "placement": "left",
              "prefixCls": "rc-tooltip",
              "trigger": Array [
                "click",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$ git init",
                "className": "line",
              },
              "ref": null,
              "rendered": "$ git init",
              "type": "p",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
